### DISCUSSION TITLE
Hide titles for stub pages automatically generated by autosummary for Python modules, classes, functions, etc.

### DISCUSSION THREAD

**Discussion by bbye98** [pradyunsg/furo#673] [Questions] [answered]
Hello!

I am currently using Sphinx and autosummary to automatically and recursively generate the API documentation for my Python project. I would like to hide the titles for the `.rst` files that are generated.

---

For example, let's say I have a module called `mdhelper.analysis.base` with the following snippet as the first lines in the source code:
```
"""
Analysis base classes (`mdhelper.analysis.base`)
================================================
This module contains...
"""
```
Currently, the HTML page will show something like

# mdhelper.analysis.base
## Analysis base classes (`mdhelper.analysis.base`)
This module contains...

However, I'd like it to not show the h1 header and show only the h2 header, similar to how the docs for NumPy (example: [numpy.linalg](https://numpy.org/doc/stable/reference/routines.linalg.html)) and SciPy (example: [scipy.optimize](https://docs.scipy.org/doc/scipy/tutorial/optimize.html)) are currently configured.

---

Similar story for classes and functions. Let's say one of my functions is `mdhelper.analysis.transport.electrophoretic_mobility`. This means that the title of the documentation page for that function looks like

# mdhelper.analysis.transport.electrophoretic_mobility
`mdhelper.analysis.transport.electrophoretic_mobility(L_ij: ndarray[float], ...)`
Calculates the...
#### PARAMETERS:
**L_ij** : _numpy.ndarray_

Again, I'd like to not have the title and have the page go directly to the function signature.

---

Is this possible?

**Comment by Eric-Arellano** [comment, marked as answer]
Hey @bbye98, have you seen that you can override the `_templates` folder for Autosummary? For example: https://github.com/Qiskit/qiskit_sphinx_theme/blob/76b3b25e04cedc947908ee8781ef8149cf6ad1c9/example_docs/docs/_templates/autosummary/class.rst?plain=1#L1-L41

You should be able to play around with things in the templates. See https://www.sphinx-doc.org/en/master/usage/extensions/autosummary.html#customizing-templates for more.

--

This feature request is not really a question of Furo. Furo has no very little control over what HTML content is generated. As a Sphinx theme, all it can do is change how things are styled via CSS and JS.

Also, you can always override Furo's styling by following the instructions at https://pradyunsg.me/furo/customisation/injecting/. For example, we change our autodoc to look like this:

https://github.com/Qiskit/qiskit_sphinx_theme/blob/main/tests/js/snapshots.test.js-snapshots/api-docs-class-page-1-linux.png

https://github.com/Qiskit/qiskit_sphinx_theme/blob/main/src/qiskit_sphinx_theme/assets/styles/_api.scss
